src/big_core_base.cpp

```diff
@@ -1,120 +1,99 @@
-// #include <cstdint> RV: smazat, toto je nutne uz v hlavicce
-
 #include "../include/big_core_base.h"
 
-// RV: general question. Proc vsude pouzivas "this->..."? Osobne mi to prijde mene prehledne a nepouzival bych to. Ale predpokladam, ze pro Tebe to bude naopak. Nebo je jiny duvod?
-// RV: general comment. Pri pouziti struktury "if (condition) statement" bud davej vse na jeden radek nebo prosim pouzivej { }. Pokud je to na vice radku bez slozenych zavorek, delam pak nekdy chybu, ze pripisu dalsi statement da dalsi radek a zapomenu pridat { }. Diky
-// RV: general comment. Vsechny komentare, ktere pravdepodobne nebudou soucasti finalni verze a jsou to spise otazky nebo odpovedi na me komentare prosim oznacuj inicialy LB, podobne jako to delam ja. Takoveto komentare mohou byt cesky bez diakritiky.
-// RV: general comment. V cpp souborech dodrzuj prosim pokud mozno stejne poradi metod jako v hlavickovem souboru. Lepe se pak hleda a kontroluje, co je naimplementovano.
-
-BigCoreBase::BigCoreBase()
-    // RV: nasledujici dva radky bych smazal, nevim proc tam cokoli nastavovat, dokud nemam parametry (rozmery, datove typy, atd.)
-	//:
-	//outermostEntitiesOffsets(0)
-{}
-
 BigCoreBase::~BigCoreBase()
 {
-    if (this->data) delete[] this->data;
-}
-
-uint64_t BigCoreBase::getNumberOfImages()
-{
-	return this->numberOfImages;
-}
-
-uint64_t BigCoreBase::getNumberOfTiles()
-{
-	return this->numberOfTiles;
-}
-
-uint64_t BigCoreBase::getImageHeight()
-{
-	return this->imageHeight;
-}
-
-uint64_t BigCoreBase::getImageWidth()
-{
-	return this->imageWidth;
-}
-
-uint64_t BigCoreBase::getNumberOfPlanes()
-{
-	return this->numberOfPlanes;
-}
-
-const std::vector<uint64_t>& BigCoreBase::getDataOrder()
-{
-    return dataOrder;
-}
-
-const std::vector<uint8_t>& BigCoreBase::getDataType()
-{
-    return dataType;
-}
-
-bool BigCoreBase::isEmpty()
-{
-	return this->dataLength == 0 ? true : false;
+    delete[] _data;
```

Comment:
Takhle nemuzes. Pokud _data nejsou nactena, tak se snazis uvolnovat nealokovanou pamet. Spravne je
```
if(this->_data)
    delete [] this->_data;
```

Reply:
Podle specifikace to může být nullptr: If this is a null-pointer, the function does nothing. Podmínka je tedy zbytečná.

Reply:
Podle jake specifikace?
```
BigCoreBase::~BigCoreBase()
{
    delete[] _data;
}
```
Tohle evidentne neco dela. Destruktor se vola pri zahazovani lokalni promenne v ramci scopu. Pokud tedy pamet nebude nactena a bude se rusit instance tridy BigCore, bude se volat destruktor a program skonci segmentation faultem.

Reply:
Pokud _data == nullptr (což se nastaví na začátku libovolného konstruktoru třídy BigCoreBase), pak delete[] _data neudělá nic, tedy žádnou chybu to nezpůsobí. Pokud _data != nullptr, pak delete[] _data uvolní paměť. Podmínka navíc by nic nového nepřidala. Specifikace například zde: http://www.cplusplus.com/reference/new/operator%20delete[]/

Reply:
priklad: vytvori se instance tridy, vse se nacte, neco se s tim udela. Budu chtit prepouzit tuhle instanci, predam novy soubor, data se nebudou vejit do pameti, _data budou uvolnena. Pri volani destruktoru segmentation fault.

Viz:
test.cpp:
```
int main()
{
    int* arr = new int[5];
    delete[] arr;
    delete[] arr;
    return 0;
}
```
`g++ -o test test.cpp -Wall -std=c++11 -pedantic -g`
./test
*** Error in `./test': double free or corruption (fasttop): 0x0000000001966010 ***
Segmentation fault (SIGABRT) (core dumped)

Reply:
Ano, tvůj test.cpp samozřejmě způsobí segmentation fault. Ale jak by se to mohlo stát s třídou Big? Pokud předám nový soubor, půjde to přes metodu openFile(fileName) (jediné další kam jde předat ten soubor, jsou metody readFile(fileName) a readFile(fileName, bytes), ale ty obě volají metodu openFile). Metoda openFile pokud soubor otevře a bude se shodovat hlavička, zavolá metodu clear(), které provede delete[] _data a nastaví _data = nullptr. Pak když se zavolá destruktor, tak to projde korektně, protože _data == nullptr. Pokud vidíš ještě jiný scénář, kdy by to mohlo skončit chybou, popiš to prosím podrobně, protože já v tom nikde problém nemůžu najít.

Reply:
mas pravdu, toho nastavovani _data = nullptr v clear() jsem si nevsim